### ISSUE TITLE
Integrate send and share in menu

### ISSUE THREAD

**Issue by jospoortvliet** [nextcloud/android#531]
### Actual behaviour
If you want to share a file over WhatsApp the 'share' menu item is of little help. Confusingly you have to use 'send'. This makes sense from a technical pov but not from and end user perspective.

### Expected behaviour
Share should allow you to share - tech details be damned. I suggest to show a share via nextcloud action as part of the standard android share menu...

**Comment by jospoortvliet** [comment]
@jancborchardt it probably needs a little thinking from you... what the logical behaviour is when sharing a file already on Nextcloud, sharing a file not on nextcloud etc...

**Comment by tobiasKaminsky** [comment]
What about "share" -> "share via nextcloud" and "send" -> "share to other app"?

**Comment by jospoortvliet** [comment]
Yeah, that could work, probably the easiest, quickest solution, though it might be too long for the menu...

The 'prettiest' solution, imho, would be to have, in the Android Share menu, one item (Nextcloud) which would act differently depending on the file you share:
* if already on Nextcloud, give the usual Nextcloud share dialog
* if not on Nextcloud, allow user to upload to Nextcloud

That way, you have only one share button/dialog.

**Comment by mario** [comment]
@tobiasKaminsky suggestion, have a share button and edit the list of available shares to add Nextcloud as a first option. That way one menu, one way to do something.

Will see what I can talk @jancborchardt into drawing :P

**Comment by tobiasKaminsky** [comment]
I do not understand your idea :/
Can you make me a mockup or find an app that is similar to the way you want to have it?

**Comment by jospoortvliet** [comment]
I don't understand it either... Is what I proposed not possible?

**Comment by jancborchardt** [comment]
I understand what @jospoortvliet means, I think. Basically, that there is not »Share« and »Send« but only »Share«. And tapping that opens the list of apps/options which are shown when you would tap »Send« – with Nextcloud / »Share via Nextcloud« being the first one. And tapping that would open what right now is visible under »Share«.

I agree this simplifies the flow because it combines two options which do the same into one.

Correct @jospoortvliet? And is this even possible @tobiasKaminsky @mario @AndyScherzinger?

If it’s not possible we could do what iOS does, calling it »Share« and renaming »Send« to »Open in …«

**Comment by jospoortvliet** [comment]
@jancborchardt you're exactly right.

**Comment by AndyScherzinger** [comment]
Imho we need to then go for the same wording as used by iOS since the two menu items do different OS calls (one asking for applications to which a file can be shared, e.g. messenger apps and one OS call asking for applications that can open/edit the file+type)

**Comment by jospoortvliet** [comment]
@AndyScherzinger makes sense. Then at least it is really 'share' and 'open in', right now it is confusing...

**Comment by jancborchardt** [comment]
Here is how Simplenote does it:
![screenshot_2017-02-09-21-13-34](https://user-images.githubusercontent.com/925062/31354417-a628e2e0-ad36-11e7-9d2b-afe409f4644d.png)
It’s one integrated Share dialog. With the Simplenote-internal share options on the top, and the typical Android "send" options at the bottom.

We could do the same for us, putting shortcuts to »Share with people« and »Share link« on the top, and the other apps on the bottom. This would clear up the confusion from our current separation between »Share« and »Send«.

What do you think @nextcloud/android?

**Comment by tobiasKaminsky** [comment]
»Share with people« and »Share link« would then both lead to the same (yet existing) share dialog?

Reference for inspiration: https://github.com/Automattic/simplenote-android/blob/74466a2b595e5e015a862460de55f0bf53c2f107/Simplenote/src/main/java/com/automattic/simplenote/ShareBottomSheetDialog.java#L38

**Comment by jancborchardt** [comment]
> »Share with people« and »Share link« would then both lead to the same (yet existing) share dialog?
> Maybe we can/should split this then up on a later iteration?

Yep, absolutely. For the first iteration a simple »Share« with the Nextcloud logo above would work. :)

**Comment by tobiasKaminsky** [comment]
@jancborchardt during hackweek we brainstormed a bit about this.
The idea was to integrate "open with" also in this dialog.

But how do we want to name the menu entry? 
Currently (on a technically point of view):
- send: send to a receipt (whats app, email, ...)
- share: share within NC
- open with: open with an *app*

(amaze as an example has also "open with" and "share")

**Comment by jancborchardt** [comment]
Maybe let’s first do the combination of Share (within Nextcloud) and Send, into »Share«.

The »Open with« we could put separately for now because it’s not as confusing, and actually might be different enough to cause confusion if we combine it.

So in any case – step by step. :)

**Comment by tobiasKaminsky** [comment]
Click on a folder:
![2017-11-16-132920](https://user-images.githubusercontent.com/5836855/32891268-33652a0c-cad2-11e7-9ec3-c4a403d24e72.png)


Click on a file:
![2017-11-16-132911](https://user-images.githubusercontent.com/5836855/32891274-36ffc960-cad2-11e7-8cbf-353fe8e292a3.png)

With #579 we lost the ability to send multiples files.
Shall I re-implement it with this PR as I changed a lot of code regarding this.
Sharing multiple files via NC is not possible.
As this only affects multi-select I would simply add "send" again to the overflow menu.

**Comment by AndyScherzinger** [comment]
I'd say add a "Share" headline and use the labels "via link" and "with people" plus add a 16dp padding between the two items (value available via ```standard_padding```).

As for re-implementing sharing of multiple files at once either do it in the PR or via another one stacked on top of this one. I'd leave the decision to you depending on the efforts necessary so this PR doesn't get slowed down or overly complex. :)

**Comment by tobiasKaminsky** [comment]
Do you want to help regarding the first design ideas?

Regarding "sharing of multiple files" I will do a PR based on this one.

**Comment by jancborchardt** [comment]
Nice! Some design feedback:

- Is the view custom? The one used by Simplenote and the one in Signal are panes coming in from the bottom, which seems like some sort of standard?
- We can simplify the wording to »Share link«. Maybe even to simply »Share« and »Link«?
- Some more whitespace between the »Share link« and »Share with people« is needed because the text currently sticks together
- The black icons are a bit heavy, let’s put them at 70% opacity.
- The divider line is also too heavy, should be 30% grey. Also check the Simplenote screenshot.
- The app titles like »Messen_ger«, »Hangout_s« and »Nextclou_d« are broken in ways which should not happen. Breaks should only happen in whitespaces

**Comment by tobiasKaminsky** [comment]
Apparently we never had "sharing of multiple files".
This is also a little bit trickier if not all files are downloaded and if they have different mimeTypes, so I'll postpone this.

**Comment by tobiasKaminsky** [comment]
![2017-11-17-092852](https://user-images.githubusercontent.com/5836855/32937673-c51296ae-cb79-11e7-93a8-ee70f8b2cb36.png)

**Comment by AndyScherzinger** [comment]
Looks nice! 👍 
@tobiasKaminsky @jancborchardt what do you think about having a circular, blue background for the icons and paint them white (like simplenote) and have it themeable too with switching the icon balck/white like we do with text in some areas.

**Comment by tobiasKaminsky** [comment]
Totally fine with this, @AndyScherzinger

**Comment by jancborchardt** [comment]
Looks great! @AndyScherzinger's suggestion also sounds good - let's use Nextcloud blue #0082c9 for that. :)

**Comment by jospoortvliet** [comment]
Looking good 😛

**Comment by andreachampion** [comment]
Love all the info

**Comment by AndyScherzinger** [comment]
Merged to master
